src/default.nix

```diff
@@ -100,6 +100,66 @@ let
       };
       parsec = import "${externalSources.nix-parsec}/parsec.nix";
     };
+    crane =
+      let
+        nix-std = import "${externalSources.nix-std}/default.nix";
+        importLibFile = name: import "${externalSources.crane}/lib/${name}.nix";
+        makeHook = name:
+          pkgs.makeSetupHook
+            ({ inherit name; } // (
+              lib.optionalAttrs (
+                name == "inheritCargoArtifactsHook"
+                || name == "installCargoArtifactsHook"
+              ) {
+                substitutions = {
+                  zstd = "${pkgs.pkgsBuildBuild.zstd}/bin/zstd";
+                };
+              }
+            ))
+            "${externalSources.crane}/pkgs/${name}.sh";
+
+        hooks =
+          lib.genAttrs [
+            "configureCargoCommonVarsHook"
+            "configureCargoVendoredDepsHook"
+            "inheritCargoArtifactsHook"
+            "installCargoArtifactsHook"
+            "remapSourcePathPrefixHook"
+          ] makeHook;
+      in rec {
+        # These aren't used by dream2nix
+        crateNameFromCargoToml = null;
+        vendorCargoDeps = null;
+
+        writeTOML = importLibFile "writeTOML" {
```

Comment:
Oh, I see what you mean. We could just put our own `writeTOML` here. That would probably be the best option.